source/common/http/filter_manager.h

```diff
@@ -1127,6 +1138,8 @@ class FilterManager : public ScopeTrackedObject,
     bool is_grpc_request_ : 1;
     // Tracks if headers other than 100-Continue have been encoded to the codec.
     bool non_100_response_headers_encoded_ : 1;
+    // True if we should rewrite the upstream response using local_reply_
+    bool do_response_rewrite_ : 1;
```

Comment:
We could probably move this state bit into `local_reply_data_` (perhaps renamed as local_reply_context_/state_ or whatever) but this approach worked quickly and IIUC there should already be a bit available in this struct without causing an extra byte to the struct space.